### PR TITLE
Add pytorch notebook

### LINKED ISSUE #476: Create example notebook for PyTorch usage
Example [code](https://github.com/chanzuckerberg/cellxgene-census/blob/main/api/python/notebooks/ml_demo/pytorch_lr_classifier.py) for a LR model training has been added to the repo, but it needs to be converted to a "proper" Jupyter notebook.

### PR DESCRIPTION
**Pull request by atolopko-czi**
Closes #476 

* add pytorch notebook
* minor pytorch api docstring updates

### CODE REVIEW

File: api/python/notebooks/experimental/pytorch.ipynb [hunk 1]
```diff
@@ -0,0 +1,675 @@
...
+   "cell_type": "code",
+   "execution_count": 2,
+   "id": "54896e6f",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import cellxgene_census.experimental.ml as census_ml\n",
+    "import tiledbsoma as soma\n",
+    "\n",
+    "experiment_datapipe = census_ml.ExperimentDataPipe(\n",
```
Comment (bkmartinjr) [inline api/python/notebooks/experimental/pytorch.ipynb:72]: I think this code would be clearer - given the prose - if it looked like:
```
   experiment = census["census_data"]["homo_sapiens"]
   experiment_datapipe = census_ml.ExperimentDataPipe(
        experiment,
        ...
    )
```

File: api/python/notebooks/experimental/pytorch.ipynb [hunk 2]
```diff
@@ -0,0 +1,675 @@
...
+    "\n",
+    "The constructor only requires a single parameter, `experiment`, which is a `soma.Experiment` containing the data of the organism to be used for training.\n",
+    "\n",
+    "To retrieve a subset of the Experiment's data, along either the `obs` or `var` axes, you may specify query filters via the `obs_query` and `var_query` parameters, which are both `soma.AxisQuery` objects.\n",
+    "\n",
+    "The values for the prediction label(s) that you intend to use for training are specified via the `obs_column_names` array. \n",
+    "\n",
+    "Finally, the `batch_size` allows you to specify the number of cells to return in each training iteration step. You may exclude this parameter if you want single rows (`batch_size=1`).\n",
```
Comment (bkmartinjr) [inline api/python/notebooks/experimental/pytorch.ipynb:95]: does batch size control matrix cells, or matrix rows.  The use of the term `cells` is confusing

**Review by bkmartinjr — APPROVED** [review]
Looks completely sufficient to release IMHO.  A couple of tiny nits noted.

File: api/python/notebooks/experimental/pytorch.ipynb (hunk 1 above)
Comment (atolopko-czi) [reply api/python/notebooks/experimental/pytorch.ipynb:74]: Agreed, that's clearer! Fixed.

File: api/python/notebooks/experimental/pytorch.ipynb (hunk 2 above)
Comment (atolopko-czi) [reply api/python/notebooks/experimental/pytorch.ipynb]: updated to say `number of obs rows (cells)`

File: api/python/notebooks/experimental/pytorch.ipynb [hunk 3]
```diff
@@ -0,0 +1,514 @@
...
+    "    batch_size=16,\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "6c7c17c3",
+   "metadata": {},
+   "source": [
+    "This class provides an implementation of PyTorch's `DataPipe` interface, which defines a common mechanism for wrapping and accessing training data from any underlying source. The `ExperimentDataPipe` class encapsulates the details of querying and retrieving Census data from a single SOMA `Experiment` and returning it to the caller as PyTorch Tensors. Most importantly, it retrieves the data lazily from the Census in batches, avoiding having to load the entire training dataset into memory at once.\n",
```
Comment (pablo-gar) [inline api/python/notebooks/experimental/pytorch.ipynb:89]:
```suggestion
    "This class provides an implementation of PyTorch's [`DataPipe` interface](https://pytorch.org/data/main/torchdata.datapipes.iter.html), which defines a common mechanism for wrapping and accessing training data from any underlying source. The `ExperimentDataPipe` class encapsulates the details of querying and retrieving Census data from a single SOMA `Experiment` and returning it to the caller as PyTorch Tensors. Most importantly, it retrieves the data lazily from the Census in batches, avoiding having to load the entire training dataset into memory at once.\n",
```

File: api/python/notebooks/experimental/pytorch.ipynb [hunk 4]
```diff
@@ -0,0 +1,514 @@
...
+  {
+   "cell_type": "markdown",
+   "id": "9c8899e7",
+   "metadata": {},
+   "source": [
+    "## Training a PyTorch Model\n",
+    "\n",
+    "This tutorial shows how to train a Logistic Regression model in PyTorch using the Census API's `experimental.ml.ExperimentDataPipe` class. This is intended only to demonstrate the use of the `ExperimentDataPipe`, and not as an example of how to train a biologically useful model.\n",
+    "\n",
+    "This tutorial assumes a basic familiarity with PyTorch and the Census API. See the [`Querying and fetching the single-cell data and cell/gene metadata`](../api_demo/census_query_extract.ipynb) notebook tutorial for a quick primer on Census API usage.\n",
```
Comment (pablo-gar) [inline api/python/notebooks/experimental/pytorch.ipynb:12]:
```suggestion
    "This tutorial assumes a basic familiarity with PyTorch and the Census API. See the [Querying and fetching the single-cell data and cell/gene metadata](https://chanzuckerberg.github.io/cellxgene-census/notebooks/api_demo/census_query_extract.html) notebook tutorial for a quick primer on Census API usage.\n",
```

needs full link or else will fail in doc-site

**Review by pablo-gar — APPROVED** [review]
Needs to be re-run, the last few cell were not executed

**Pull request merged**